public/js/app.Ctrl.js

```diff
@@ -0,0 +1,201 @@
+App.controller('App.Ctrl', function($scope, $timeout, AppService) {
+    const colorCode = {1:"green", 2: "red", 3: "yellow", 4: "blue"};
+    const speed = 500; // in milliseconds
+    var on = false;
```

Comment:
Start using some emca6 syntax.  In general using 'let' is superior to 'var', read about the differences as you will probably be tested on stuff like that.  When you have a ton of variables like this, its definitely a sign that you can change your app logic somewhat to get rid of some of them.  Even then, if you have consecutive var declarations, use one var, and separate the rest with commas, with one variable per line, indenting all but the first one.